internal/resources/collection.go

```diff
@@ -0,0 +1,404 @@
+	return &Collection{}
+}
+
+// ImportState imports a remote collection that is not created by Terraform.
+func (c *Collection) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import name and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("collection_name"), req, resp)
+}
+
+// Metadata returns the Collection resource type name.
+func (c *Collection) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_collection"
+}
+
+// Schema defines the schema for the Collection resource.
+func (c *Collection) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = CollectionSchema()
+}
+
+// Configure It adds the provider configured api to the collection resource.
+func (c *Collection) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	c.Data = data
+}
+
+// Create creates a new collection.
+func (c *Collection) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.Collection
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	collectionRequest := collection_api.CreateCollectionRequest{
+		Name: plan.Name.ValueString(),
+	}
+
+	// Check for optional fields
+	if !plan.MaxTTL.IsNull() && !plan.MaxTTL.IsUnknown() {
+		collectionRequest.MaxTTL = plan.MaxTTL.ValueInt64Pointer()
+	}
+
+	if err := c.validateCreateCollectionRequest(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error parsing create collection request",
+			"Could not create collection, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	var organizationId = plan.OrganizationId.ValueString()
+	var projectId = plan.ProjectId.ValueString()
+	var clusterId = plan.ClusterId.ValueString()
+	var bucketId = plan.BucketId.ValueString()
+	var scopeName = plan.ScopeName.ValueString()
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/buckets/%s/scopes/%s/collections", c.HostURL, organizationId, projectId, clusterId, bucketId, scopeName)
+	cfg := collection_api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusCreated}
+	_, err := c.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		collectionRequest,
+		c.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating collection",
+			errorMessageWhileCollectionCreation+collection_api.ParseError(err),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, plan)
```

Comment:
"Since we have made maxttl as computed, while populating the plan, we need to check if the maxttl is not provided. If it is not provided in the plan, then we should set it to null."